packages/client/src/types.ts

```diff
@@ -31,16 +32,42 @@ export type AddressType = {
 
 export type Address = null | AddressType;
 
+/**
+ * Section types
+ */
+
+export type SectionComponent = {
+  currentSection: SectionObject;
+  sectionFunctions: SectionFunctions;
+};
+
+type SectionData = {
+  index: number;
+  isActive: boolean;
+  isCompleted: boolean;
+};
+
+export type SectionObject = SectionData & {
+  component: (props: SectionComponent) => {};
+};
+
+export type SectionFunctions = {
+  activateSection: (section: SectionObject) => void;
+  completeSection: (state?: boolean, section?: SectionObject | null) => void;
+  getNextSection: () => SectionObject | null;
+  goToNextSection: () => void;
+};
+
 /**
  * Context and session types
  */
 export type TopicData = {
-  activeComponents?: string[];
   address: Address;
   answers: {
     [id: string]: Answer;
   };
-  finishedComponents: string[];
+  timesCheckerLoaded: number;
```

Comment:
Can't this be a boolean? checkerLoaded (true / false)?

Reply:
No unfortunately I need to know if `timesCheckerLoaded === 0` (using the NewCheckerModal) `timesCheckerLoaded === 1` (on first load via normal route from Intro Page) or `timesCheckerLoaded > 1` (when the page reloaded). So a boolean doesn't do. But I'm very open to discuss other options because I don't really like this way either.

Reply:
I've added this topic to next dev-meeting

Reply:
Resolving this for now